test/config/utility.h

```diff
@@ -104,17 +104,10 @@ class ConfigHelper {
   // Set the connect timeout on upstream connections.
   void setConnectTimeout(std::chrono::milliseconds timeout);
 
-  // TODO(alyssawilk) this does not scale. Refactor.
-  // Add an additional route to the configuration.
-  void addRoute(const std::string& host, const std::string& route, const std::string& cluster,
-                bool validate_clusters,
-                envoy::api::v2::route::RouteAction::ClusterNotFoundResponseCode code,
-                envoy::api::v2::route::VirtualHost::TlsRequirementType type =
-                    envoy::api::v2::route::VirtualHost::NONE,
-                envoy::api::v2::route::RetryPolicy retry_policy = {},
-                bool include_attempt_count_header = false, const absl::string_view upgrade = "",
-                const envoy::api::v2::route::RouteAction::InternalRedirectAction internal_action =
-                    envoy::api::v2::route::RouteAction::PASS_THROUGH_INTERNAL_REDIRECT);
+  envoy::api::v2::route::VirtualHost createHost(const char* host, const char* route = "/",
+                                                const char* cluster = "cluster_0");
+
+  void addVirtualHost(envoy::api::v2::route::VirtualHost& vhost);
```

Comment:
any reason we're not passing this by const ref?

Reply:
Oh, I'd waffled about using Swap instead of CopyFrom and then forgot so got worst of both :-P
Good catch!